guardian/gudocs
# Fix parsing bug, update dependencies & docs

## Reason for change

A problem occurred with a google sheet failing to parse correctly. 

## Changes

1. Update parsing library
Since all CSV -> JSON parsing in gudocs is done by a library `Babyparse`, we decided that update to the latest version of the library was the best thing to do. Since `Babyparse` [has been deprecated](https://github.com/Rich-Harris/BabyParse) in favour of `Papaparse` we installed that instead. By testing on the [library demo website](https://www.papaparse.com/demo), it seemed that the CSV could be parsed fine by the latest version. 


2. Update Grunt, Grunt-Sass and the Node version 
The Papaparse module caused problems with some of the older versions of npm module that we had (particularly Grunt-Sass), so I have updated as many of these as I could without causing an error. Node has been bumped to 8.5.0 and set with an `.nvmrc` file. 

3. Update documentation and loading scripts
The documentation had got out of date, particularly around getting the correct credentials from Google. They now run nvm when either of the dev scripts are run. And the run locally script also runs redis automatically

## To Test
Get credentials and run `./dev-start-ui.sh` and then `./dev-start-fetch.sh`. 

The dev account should have a sheet called `Copy 2 docstool bug reproduced` shared with it. 

When you run fetch script this should end up in this bucket on composer: `gudocs-dev/docsdata-test/` and the tab called `Parsing Error Here` should be parsed correctly
